pkg/oc/cli/cmd/newapp.go

```diff
@@ -154,6 +156,50 @@ type NewAppOptions struct {
 	*ObjectGeneratorOptions
 }
 
+func nonEmpty(str string, def string) string {
+	if str == "" {
+		return def
+	}
+	return str
+}
+
+func checkResources(discoveryClient discovery.DiscoveryInterface, items []runtime.Object) error {
+	resources, err := discoveryClient.ServerResources()
+	if err != nil {
+		return fmt.Errorf("Unable to to get list of available resources: %v", err)
+	}
+
+	resourceMap := make(map[schema.GroupVersionKind]bool)
+	for _, resourceList := range resources {
+		for _, resource := range resourceList.APIResources {
+			listGv, _ := schema.ParseGroupVersion(resourceList.GroupVersion)
+			gvk := schema.GroupVersionKind{
+				Group:   nonEmpty(resource.Group, listGv.Group),
+				Version: nonEmpty(resource.Version, listGv.Version),
+				Kind:    resource.Kind,
+			}
+			resourceMap[gvk] = true
+		}
+	}
+
+	declaredResources := make(map[schema.GroupVersionKind]bool)
+	for _, item := range items {
+		versioned := kcmdutil.AsDefaultVersionedOrOriginal(item, nil)
+		declaredResources[versioned.GetObjectKind().GroupVersionKind()] = true
+	}
+	missingResources := make([]string, 0)
+	for r := range declaredResources {
+		if present := resourceMap[r]; !present {
+			missingResources = append(missingResources, r.String())
+		}
+	}
+
+	if len(missingResources) > 0 {
+		return fmt.Errorf("Missing declared resources: %v", strings.Join(missingResources, `, `))
```

Comment:
"API server is missing declared resources:"

Reply:
corrected, thanks!